src/core/metrics.cc

```diff
@@ -132,18 +177,18 @@ Metrics::HashLabels(const std::map<std::string, std::string>& labels)
 
 Metrics::~Metrics()
 {
+  // Signal the cache thread to exit and then wait for it...
+  if (poll_thread_ != nullptr) {
+    poll_thread_exit_.store(true);
+    poll_thread_->join();
 #ifdef TRITON_ENABLE_METRICS_GPU
-  // Signal the DCGM thread to exit and then wait for it...
-  if (dcgm_thread_ != nullptr) {
-    dcgm_thread_exit_.store(true);
-    dcgm_thread_->join();
-    dcgmGroupDestroy(dcgm_handle_, groupId_);
+    dcgmGroupDestroy(dcgm_metadata_.dcgm_handle_, dcgm_metadata_.groupId_);
```

Comment:
Should check if meta data is initialized and clean it up only if set. Or you may make meta data a class and do it in the destructor

Reply:
done